File: portefeuille_rb/wallet.rb
class PortefeuilleVirtuel
    def initialize
      @solde = 0
    end
  
    def afficher_solde
      puts "Solde actuel : #{@solde} euros"
    end
  
    def ajouter_argent
      print "Entrez le montant à ajouter : "
      montant = gets.chomp.to_f
  
      if montant > 0
        @solde += montant
        puts "Vous avez ajouté #{montant} euros à votre portefeuille."
      else
        puts "Veuillez entrer un montant valide."
      end
    end
  
    def retirer_argent
      print "Entrez le montant à retirer : "
      montant = gets.chomp.to_f
  
      if 0 < montant && montant <= @solde
        @solde -= montant
        puts "Vous avez retiré #{montant} euros de votre portefeuille."
      else
        puts "Veuillez entrer un montant valide ou vérifier votre solde."
      end
    end
  
    def menu_principal
      loop do
        puts "\nMenu principal :"
        puts "1. Afficher le solde"
        puts "2. Ajouter de l'argent"
        puts "3. Retirer de l'argent"
        puts "0. Quitter"
  
        print "Choisissez une option : "
        choix = gets.chomp
  
        case choix
        when "1"
          afficher_solde
        when "2"
          ajouter_argent
        when "3"
          retirer_argent
        when "0"
          puts "Merci, au revoir !"
          break
        else
          puts "Option invalide. Veuillez réessayer."
        end
      end
    end
  end
  
  # Lancement du programme
  portefeuille = PortefeuilleVirtuel.new
  portefeuille.menu_principal
  

File: portefeuille_js/wallet.js
const readlineSync = require('readline-sync');

let solde = 0;

function afficherSolde() {
    console.log(`Solde actuel : ${solde} euros`);
}

function ajouterArgent() {
    const montant = parseFloat(readlineSync.question('Entrez le montant à ajouter : '));
    if (!isNaN(montant) && montant > 0) {
        solde += montant;
        console.log(`Vous avez ajouté ${montant} euros à votre portefeuille.`);
    } else {
        console.log('Veuillez entrer un montant valide.');
    }
}

function retirerArgent() {
    const montant = parseFloat(readlineSync.question('Entrez le montant à retirer : '));
    if (!isNaN(montant) && montant > 0 && montant <= solde) {
        solde -= montant;
        console.log(`Vous avez retiré ${montant} euros de votre portefeuille.`);
    } else {
        console.log('Veuillez entrer un montant valide ou vérifier votre solde.');
    }
}

function menuPrincipal() {
    while (true) {
        console.log('\nMenu principal :');
        console.log('1. Afficher le solde');
        console.log('2. Ajouter de l\'argent');
        console.log('3. Retirer de l\'argent');
        console.log('0. Quitter');

        const choix = parseInt(readlineSync.question('Choisissez une option : '));

        switch (choix) {
            case 1:
                afficherSolde();
                break;
            case 2:
                ajouterArgent();
                break;
            case 3:
                retirerArgent();
                break;
            case 0:
                console.log('Merci, au revoir !');
                process.exit(0);
            default:
                console.log('Option invalide. Veuillez réessayer.');
        }
    }
}

menuPrincipal();
